### ISSUE TITLE
Task combination in a single model

### ISSUE THREAD

**Issue by tschimbr** [eonum/medcodelearn#7]
Combine the different tasks (PDX, SDX, SRG and DRG) in a single model with shared input and intermediate layers.

**Comment by tschimbr** [comment]
Create a softmax group for each task in the output layer

**Comment by tschimbr** [comment]
Use Graph

**Comment by tschimbr** [comment]
An alternative would be to use the weights from other task from all layers except the output layer as a sort of pre training. Though this would not have the advantage of having a single model that does not need much space and computational resources.

**Comment by tschimbr** [comment]
Probably it will not be possible to use different target labels for different tasks for the same training sample. (Check if it is possible to encode missing targets in Keras?) But it should be possible to do this for test samples.

**Comment by tschimbr** [comment]
Wait until Keras 1.0 is out with the new Graph API.

**Comment by tschimbr** [comment]
Keras is upgraded to 1.0. LSTM is now implemented using the functional API.
Pretraining parts of the model with other tasks is also implemented in https://github.com/eonum/medcodelearn/tree/reuse-model. Though accuraccy does not improve (~1.0% worse than without usage of pretrained model)
